firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapEncoder.kt

```diff
@@ -0,0 +1,208 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx.serialization
+
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.builtins.LongAsStringSerializer.descriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.StructureKind
+import kotlinx.serialization.encoding.AbstractEncoder
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.modules.EmptySerializersModule
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.serializer
+
+/**
+ * The entry point of Firestore Kotlin Serialization Process. It encodes the @[Serializable] objects
+ * into nested maps of Firestore supported types.
+ *
+ * For a @[Serializable] object:
```

Comment:
Nit: do you know if this is going to render as a list in kotlindoc? Since it's a private method is not a big deal if not, but if there's not much work to get it to work would be nice.

Reply:
Yes, I have tried to convert a Java file to Kotlin, and this is how the automatically converted code looks like.